File: api/src/main/java/com/hirrao/javaee/mapper/BloodPressureTableMapper.java
package com.hirrao.javaee.mapper;

import com.hirrao.javaee.entity.BloodPressureTable;
import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import org.apache.ibatis.annotations.Select;

import java.time.LocalDate;

public interface BloodPressureTableMapper extends BaseMapper<BloodPressureMapper> {
    @Select("SELECT AVG(SBP) as avgSBP, MIN(SBP) as minSBP, MAX(SBP) as maxSBP, AVG(DBP) as avgDBP, MIN(DBP) as minDBP, MAX(DBP) as maxDBP, AVG(SBP-DBP) as avgPP, MIN(SBP-DBP) as minPP, MAX(SBP-DBP) as maxPP FROM bloodPressureLog WHERE uid= #{uid} AND DATE(recordTime) BETWEEN DATE_SUB(#{date}, INTERVAL 6 DAY)  AND #{date};")
    BloodPressureTable getBloodPressureTable(LocalDate date, Long uid);
}

File: api/src/main/kotlin/cn/hirrao/javaee/mapper/BloodPressureTableMapper.kt
package cn.hirrao.javaee.mapper

import cn.hirrao.javaee.entity.BloodPressureTable
import com.baomidou.mybatisplus.core.mapper.BaseMapper
import org.apache.ibatis.annotations.Select
import java.time.LocalDate

interface BloodPressureTableMapper : BaseMapper<BloodPressureMapper?> {
    @Select("SELECT AVG(SBP) as avgSBP, MIN(SBP) as minSBP, MAX(SBP) as maxSBP, AVG(DBP) as avgDBP, MIN(DBP) as minDBP, MAX(DBP) as maxDBP, AVG(SBP-DBP) as avgPP, MIN(SBP-DBP) as minPP, MAX(SBP-DBP) as maxPP FROM bloodPressureLog WHERE uid= #{uid} AND DATE(recordTime) BETWEEN DATE(?, '-6 days') AND DATE(?);;")
    fun getBloodPressureTable(date: LocalDate?, uid: Int): BloodPressureTable?
}
